### ISSUE TITLE
Consider writing a detekt-internal rule for RuleNameIsTooLongAndThereforeShouldNotGetReleased

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#5124]
## Context
Looking at the new rule `DestructuringDeclarationWithTooManyEntries` in context, it really stands out:
<img height=200px src=https://user-images.githubusercontent.com/2906988/180611838-fa160452-5916-4004-a293-e9caead25171.png />

`@Suppress("DestructuringDeclarationWithTooManyEntries")` is 64 characters long, so it might be too long for some code styles. :)

## Expected Behavior of the rule
Detekt on the detekt codebase should fail PRs with long names.

Each name should be easily remembered and easy to type for a better user (dev) experience.

**Comment by 3flex** [comment]
I don't know about this, if there's a specific issue that this is causing we can take a look and consider renaming the rule, but in general I think contributors aim for succinct rule names where possible, and maintainers consider the implications of proposed new rule names before accepting contributions.

This particular rule name doesn't seem excessive to me, and truncating or shortening it may then trigger feedback that the rule name is not clear.

I'll leave it open for other feedback.

**Comment by BraisGabin** [comment]
I prefer to have a clear name than a shorter name too. If the `@Suppress` is too long you can add an `alias` to the rule and use it instead.

I could also consider a rename if something clearly better is proposed but I don't think we need a rule for that.

**Comment by TWiStErRob** [comment]
It's really long, if you try to use it in a sentence or even try to talk about it (e.g. in documentation, or comments) you'll very quickly run into problems with understandability or style. Really, I challenge you to talk about this rule in a voice call 🤓.

For example compare these two paragraphs:

> We use `MagicNumber` and `ForbiddenImport` to get a consistently readable and hopefully less buggy application. `MagicNumber` flags things that would be confusing to read and forces us to think about making a name for a thing; and `ForbiddenImport` documents our choices of libraries and conventions.

> We can try to improve your style, Detekt's new `DestructuringDeclarationWithTooManyEntries`, `UseRequire` and `UseCheckOrError` will give you hints wherever we're using confusing structures.  `DestructuringDeclarationWithTooManyEntries` will flag too many component usages, as the order can change and break or code, let's prefer simpler `x.val` calls; the other `Use...` checks will flag unnecessarily used and thrown exceptions.

One thing to note is that I was trying to write this naturally and for the others I typed out the rule names, while this long one I had to copy-paste. Anther thing is that I can't write a nicely formed flowing sentence/paragraph because the "word" is just too long and wraps strangely.

---

Additionally when suppressed there's way less space for explanations:
```kotlin
@Suppress(
    "MagicNumber", // It's obvious what 1337 is, isn't it?
    "DestructuringDeclarationWithTooManyEntries", // In this case it's still acceptable, Vector4 structure will never change.
)
```

---

To suggest a better name: I would remove the filler words beating around the bush and use an English word that depicts the same "with too many entries" concept: `LengthyDestructuring`. Note that "Destructuring Declaration" feels in itself redundant similar to "DVD Disk", because a destructuring is always at a declaration (be it local variable or lambda parameter). Yes, there are other types of declarations, but "destructuring" implies declaration.

---

I think the point here is that I think there should be more challenging in the naming to find a better set of names that can be used by developer in speech and easily remembered. I would draw the line at most 4 words and a hard limit of 30 on top of this. Of course if there's no better alternative this can be suppressed as usual, but calling it out automatically and maybe documenting how to create more succinct and clear names in context would help to move the ecosystem forward. Naming is hard, yes, but also very important as public names are used in contexts you wouldn't imagine.

**Comment by cortinico** [comment]
> If the `@Suppress` is too long you can add an `alias` to the rule and use it instead.

+1 for adding `LengthyDestructuring` as an alias for this rule 👍

**Comment by BraisGabin** [comment]
I'm closing this issue as won't do. I mean, we don't want a "max lenght" in our names. But I agree with many of the things that you pointed our @TWiStErRob. I would consider an issue to fix the name of `DestructuringDeclarationWithTooManyEntries` and change it for another one.  Lately there are some issues were we lack the option to deprecate a rule so if we implement it we could use it to fix this kind of things too.

**Comment by TWiStErRob** [comment]
Fair enough, feel free to move this as a discussion too. Thanks for listening.
